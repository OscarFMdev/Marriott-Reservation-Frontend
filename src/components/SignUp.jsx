/* eslint-disable camelcase */
import React, { useState, useEffect } from 'react';
import { InputText } from 'primereact/inputtext';
import { Password } from 'primereact/password';
import { Button } from 'primereact/button';
import { NavLink, useNavigate } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { signUp } from '../redux/reducer/Auth/auth';

const SignUp = () => {
  const navigate = useNavigate();
  const [name, setName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [password_confirmation, setPasswordConfirmation] = useState('');
  const dispatch = useDispatch();
  const { status, message } = useSelector((state) => state.auth);

  const handleSubmit = (e) => {
    e.preventDefault();
    const user = {
      name,
      email,
      password,
      password_confirmation,
    };
    dispatch(signUp(user));
  };

  useEffect(() => {
    if (message === 'Sign up success') {
      navigate('/login');
    }
  }, [message, navigate]);

  return (
    <div className="p-grid p-justify-center">
      <div className="p-col-12 p-md-8 p-lg-6">
        <div className="card">
          { status === 'failed' && <p>{message}</p> }
          <h1>Sign Up</h1>
          <form onSubmit={handleSubmit}>
            <div className="p-field pb-2">
              <span className="p-input-icon-right">
                <i className="pi pi-user" />
                <InputText
                  id="name"
                  type="text"
                  value={name}
                  onChange={(e) => setName(e.target.value)}
                  label="Name"
                  placeholder="Full Name"
                  required
                />
              </span>
            </div>

            <div className="p-field pb-2">
              <span className="p-input-icon-right">
                <i className="pi pi-envelope" />
                <InputText
                  id="email"
                  type="email"
                  value={email}
                  onChange={(e) => setEmail(e.target.value)}
                  label="Email"
                  placeholder="Email"
                  required
                />
              </span>
            </div>

            <div className="p-field pb-2">
              <Password
                id="password"
                value={password}
                onChange={(e) => setPassword(e.target.value)}
                label="password"
                placeholder="Password"
                required
                toggleMask
              />
            </div>

            <div className="p-field pb-2">
              <Password
                id="password_confirmation"
                value={password_confirmation}
                onChange={(e) => setPasswordConfirmation(e.target.value)}
                label="passconf"
                placeholder="Password"
                required
                feedback={false}
                toggleMask
              />
            </div>

            <Button label="Sign Up" type="submit" className="bg-green-500 border-200" />
            <p>
              Already have an account?
              <NavLink to="/login">Login</NavLink>
            </p>
          </form>
        </div>
      </div>
    </div>
  );
};

export default SignUp;
