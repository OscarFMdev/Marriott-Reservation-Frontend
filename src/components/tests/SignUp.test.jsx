import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import SignUp from '../SignUp';
import store from '../../redux/configureStore';

describe('It displays the Sign Up two times', () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <Router>
          <SignUp />
        </Router>
      </Provider>,
    );
  });

  test('Sign Up title and button', () => {
    expect(screen.queryAllByText(/Sign Up/, { exact: true })).toHaveLength(2);
  });
});

test('Sign Up renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <SignUp />
      </Router>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
