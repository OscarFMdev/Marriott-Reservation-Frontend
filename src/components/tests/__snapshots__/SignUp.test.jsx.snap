// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sign Up renders correctly 1`] = `
<div
  className="p-grid p-justify-center"
>
  <div
    className="p-col-12 p-md-8 p-lg-6"
  >
    <div
      className="card"
    >
      <h1>
        Sign Up
      </h1>
      <form
        onSubmit={[Function]}
      >
        <div
          className="p-field pb-2"
        >
          <span
            className="p-input-icon-right"
          >
            <i
              className="pi pi-user"
            />
            <input
              className="p-inputtext p-component"
              id="name"
              label="Name"
              onChange={[Function]}
              onInput={[Function]}
              onKeyDown={[Function]}
              onPaste={[Function]}
              placeholder="Full Name"
              required={true}
              type="text"
              value=""
            />
          </span>
        </div>
        <div
          className="p-field pb-2"
        >
          <span
            className="p-input-icon-right"
          >
            <i
              className="pi pi-envelope"
            />
            <input
              className="p-inputtext p-component"
              id="email"
              label="Email"
              onChange={[Function]}
              onInput={[Function]}
              onKeyDown={[Function]}
              onPaste={[Function]}
              placeholder="Email"
              required={true}
              type="email"
              value=""
            />
          </span>
        </div>
        <div
          className="p-field pb-2"
        >
          <div
            className="p-password p-component p-inputwrapper p-input-icon-right"
            id="password"
            style={null}
          >
            <input
              className="p-inputtext p-component p-password-input"
              id={null}
              label="password"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onInput={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onPaste={[Function]}
              placeholder="Password"
              required={true}
              style={null}
              type="password"
              value=""
            />
            <i
              className="pi pi-eye"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="p-field pb-2"
        >
          <div
            className="p-password p-component p-inputwrapper p-input-icon-right"
            id="password_confirmation"
            style={null}
          >
            <input
              className="p-inputtext p-component p-password-input"
              id={null}
              label="passconf"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onInput={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onPaste={[Function]}
              placeholder="Password"
              required={true}
              style={null}
              type="password"
              value=""
            />
            <i
              className="pi pi-eye"
              onClick={[Function]}
            />
          </div>
        </div>
        <button
          aria-label="Sign Up"
          className="p-button p-component bg-green-500 border-200"
          disabled={false}
          type="submit"
        >
          <span
            className="p-button-label p-c"
          >
            Sign Up
          </span>
        </button>
        <p>
          Already have an account?
          <a
            className=""
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </p>
      </form>
    </div>
  </div>
</div>
`;
